### PR TITLE
Remove nuke launch and do some cleanup.

### PR DESCRIPTION
**Pull request by kevingranade**
#### Summary
SUMMARY: None

#### Purpose of change
Just a little cleanup, if possible I'm going to dismantle the nuke launch code.

#### Describe the solution
Shift some stuff out of the computer header file.
Remove nuke launch code.

#### Describe alternatives you've considered
Leave broken code that causes complaints in the game.

#### Testing
Need to verify that this doesn't break saves, might require some migration.

### CODE REVIEW

**Comment by Night-Pryanik** [comment]
Could you please explain why are you removing this?

**Comment by thethunderhawk** [comment]
@Night-Pryanik
Actual formal Kevin explanation aside, I think that’s the sort of sophisticated engineering project that goes beyond the scope of CDDA. There are very sophisticated security systems controlling those things, it’d be a monumental effort to bypass them such that you could actually launch the missile at will. Also, the targeting programs are predefined, you can’t just point and shoot them wherever you like, you’d have to develop a new program to give them a new target. Who even knows how that’d work, these missiles are designed to follow a ballistic trajectory around the planet. Just lobbing them a few miles from the launch site might be a challenge in itself. It’d take serious engineering to get past all these issues.

Plus, an ICBM payload should realistically be causing so much destruction that it’d cause issues with how we scale the game. A town of a few hundred buildings should realistically be leveled dozens of times over. The way we scale the overmap, dropping a realistic nuke on it should cause inconveniently massive amounts of destruction, essentially rendering entire maps barren and uninhabitable. 

All that being said, it’d sure be super cool and interesting to be able to do all that stuff in a realistic manner, but the implementation we’ve got is nowhere close to that.

**Comment by tenmillimaster** [comment]
Personally I think it's a good idea because we don't do any fallout simulations post detonation of a random selection of warheads in silos. The smallest missile variant is, to my knowledge, a W80 cruise missile with a 150kt warhead.

That said, I'd also like to hear Kevin's intent.

**Comment by thethunderhawk** [comment]
This is neither-here-not-there, but the only silo-based missile in the US arsenal right now is the Minuteman-III, for which the smallest possible payload is one W-78 which was announced to be 325-350 Kt. That’s assuming they only mount 1/3 possible warheads in the missile, which is unlikely.
All that to say, good point. It’d be hard to scale that down in any satisfying manner without building a whole new system.

**Comment by Destragon** [comment]
What about mininukes?

**Comment by tenmillimaster** [comment]
@Destragon 



Nukemap's fallout model won't do sub 1kt warheads, but this is the plume for a 1kt warhead. A mininuke, the size of say a Davy Crockett warhead (20t), would have a much smaller plume.

**Comment by kevingranade** [comment]
Two angles of "why"
1. The code is currently broken as fuck, and has never worked right. It rightfully causes bug reports because it doesn't accomplish things it should, like wipe out fungal blooms or cities full of zombies.

2. I'm not interested in fixing it for reasons @thethunderhawk covered. It's not *remotely* feasible that a lone survivor or a team of them could break into a missile silo and successfully launch a nuke at a local target.

I'm interested in replacing this functionality with a similar in practice system where the survivor can subvert a bank of automated artillery at a military base and fire conventional ordinance at a distant target.

**Comment by Night-Pryanik** [comment]
Could nuke launch option be moved to Aftershock instead of removing?

**Comment by thethunderhawk** [comment]
IIRC the current nuke launch code doesn’t kill mobs or destroy buildings. Idk if it would even fit in aftershock for that reason.

**Comment by Night-Pryanik** [comment]
I personally tried to fix the bug with explosion not killing monsters, but with no luck unfortunately. It could be fixed by someone more competent if it remains in the code rather than purged.

**Comment by thethunderhawk** [comment]
Thanks for trying. You could put the code in limbo somewhere maybe someone would get around to it. Probably no reason to keep it in the game though, it seems like a bit of a long shot that anyone would fix that anytime soon.

**Comment by anothersimulacrum** [comment]
> You could put the code in limbo somewhere maybe someone would get around to it.

That's just dead code, and dead code should just be removed.

**Comment by thethunderhawk** [comment]
Yes agreed, just saying there’s plenty of space left over at Pastebin if @Night-Pryanik thinks the code is worth saving.

**Comment by kevingranade** [comment]
This just needs a final check that deprecation has been successful.
Visit a missile silo that was generated before the change and access the computer console.
Visit a missile silo that was generated after this change and access the console.

**Comment by ZhilkinSerg** [comment]
Something is not right with old savegames loaded with current PR:

There are both options (`Launch Missile` and `Disarm Missile`).
`Launch Missile` option does `Disarm Missile` action.
`Disarm Missile` option proceeds to `Bionic access - Manifest` menu.

image

Computers in newly generated silos are okay and only contain `Disarm Missile` menu.

**Comment by kevingranade** [comment]
Bah, thanks for testing, I'll look into that when I have time.

**Comment by kevingranade** [comment]
I'm a dumb, I blacklisted the name instead of the action.

**Comment by kevingranade** [comment]
Ok this actually works now, back to blacklisting the name, but correctly this time.
Also had to restore the action enum to avoid breaking the ordering since it still writes out bare enum values, need to clean that up sometime.

File: src/computer.cpp [hunk 1]
```diff
@@ -631,69 +634,6 @@ void computer::activate_function( computer_action action )
         }
         break;
```
Comment (ZhilkinSerg) [inline src/computer.cpp]: There is no `default` statement in `switch` and `COMPACT_OBSOLETE` case is missing too, so most compilers would complain about that.

```suggestion

        case COMPACT_OBSOLETE:
            break;
```

**Pull request merged**